webapp-django/crashstats/status/context_processors.py

```diff
@@ -0,0 +1,18 @@
+from crashstats.status.models import StatusMessage
+
+
+def status_message(request):
+    statuses = (
+        StatusMessage.objects.filter(enabled=True).order_by('-created_at')
+    )
+
+    if not len(statuses):
```

Comment:
Actually, it is applicable. It's always a good idea to do things right :)
What you're *currently* doing, if you deep log in the database is:
`select * from status_messages where ...`

If you first do a `.exists()` and then a select you're going to trigger 2 database queries:
`select 1 from status_messages where ...` AND `select * from status_messages where ...`.

However, the first query is fast and 99% of the time, the second query won't even have to happen. 

Also, from a performance point of view, what often takes time is the python code to "dress up" the results. I.e. making instances of the `StatusMessage` model in a list-list structure. 

MongoKit was an ORM for MongoDB. It too would have a model and when you make a query, it talks to the database and takes each python `dict` and instantiates it from a model. So, if your model has methods, you can do something like:
```python
for model_instance in model.query(...):
    model_instance.some_method()
```
Anyway, in [this blog post](https://www.peterbe.com/plog/mongokit-dict_plus) I played with opting out of model instance creation and used a DotDict-like thing instead and for a large recordset it made a certain big query 50 times faster.